helix-view/src/clipboard.rs

```diff
@@ -276,12 +276,27 @@ pub mod provider {
                 let stdin = input.map(|_| Stdio::piped()).unwrap_or_else(Stdio::null);
                 let stdout = pipe_output.then(Stdio::piped).unwrap_or_else(Stdio::null);
 
-                let mut child = Command::new(self.prg)
+                let mut command: Command = Command::new(self.prg);
+
+                let mut command_mut: &mut Command = command
                     .args(self.args)
                     .stdin(stdin)
                     .stdout(stdout)
-                    .stderr(Stdio::null())
-                    .spawn()?;
+                    .stderr(Stdio::null());
+
+                // Fix for https://github.com/helix-editor/helix/issues/5424
+                if cfg!(not(any(windows, target_os = "wasm32", target_os = "macos"))) {
+                    use std::os::unix::process::CommandExt;
+
+                    unsafe {
+                        command_mut = command_mut.pre_exec(|| match libc::setsid() {
+                            -1 => Err(std::io::Error::last_os_error()),
+                            _ => Ok(()),
+                        });
+                    }
+                }
```

Comment:
`setsid` should be called on all unix systems:

```suggestion
                #[cfg(unix)]
                unsafe {
                    use std::os::unix::process::CommandExt;
                    command_mut = command_mut.pre_exec(|| {
                        if libc::setsid() == -1 {
                            return Err(io::Error::last_os_error());
                        }
                    });
                }
```

For windows we might also want to pass `.creation_flags(CREATE_NEW_PROCESS_GROUP)`

Reply:
I've made the following changes:
- Call `setsid` for all unix systems
- Call `.creation_flags(CREATE_NEW_PROCESS_GROUP)` for windows

After making the changes I've realised the change to set `CREATE_NEW_PROCESS_GROUP` will never be executed as it is inside a `mod provider` that uses the config `#[cfg(not(target_os = "windows"))]`.

Seeking your opinion but I am thinking of reverting back by one commit so that the PR will contain no code related to setting `CREATE_NEW_PROCESS_GROUP` for windows but keeps the change you suggested of calling `setsid` for all unix systems?

Separate to this PR but I'm also considering doing a follow-up PR to cleanup the code a little, let me know if you think this would be valuable?
With the amount of conditional compilation attributes I found it slightly hard to follow what was going on andwas thinking of splitting the code out into separate files for each environment, e.g. a provider file for windows, macos, osc52, wasm, unix etc..

Reply:
I think these changes are ready for another review now.

I'm not able to change how `pre_exec` is called to exactly as you suggested because it needs to return a `Result<()>` type.